cli/src/deno.rs

```diff
@@ -199,35 +206,89 @@ impl ModuleLoader for ExtensionsModuleLoader {
             };
 
             if should_transpile {
-                code = transpile(module_specifier.to_string(), code, media_type)?;
+                let transpiled =
+                    transpiler.transpile(module_specifier.to_string(), code, media_type)?;
+                code = transpiled.text.clone();
             }
 
             Ok(ModuleSource::new(module_type, code.into(), &module_specifier))
         })
     }
 }
 
-/// Transpile code to JavaScript.
-fn transpile(
-    specifier: impl Into<String>,
-    code: impl Into<String>,
-    media_type: MediaType,
-) -> Result<String> {
-    let parsed = deno_ast::parse_module(ParseParams {
-        text_info: SourceTextInfo::from_string(code.into()),
-        specifier: specifier.into(),
-        capture_tokens: false,
-        scope_analysis: false,
-        maybe_syntax: None,
-        media_type,
-    })?;
-    Ok(parsed.transpile(&Default::default())?.text)
+impl SourceMapGetter for ExtensionsModuleLoader {
+    fn get_source_map(&self, file_name: &str) -> Option<Vec<u8>> {
+        let transpiled = self.transpiler.transpiled_cache.get(file_name)?;
+        transpiled.source_map.clone().map(|map| map.into_bytes())
+    }
+
+    fn get_source_line(&self, file_name: &str, line_number: usize) -> Option<String> {
+        let source = self.transpiler.source_cache.get(file_name)?;
```

Comment:
Unfortunately, this fails for any source that we didn't need to transpile (i.e., JavaScript files)... So for those files, we would need to cache the source with no source map.

You can see an example of this regression here:

```sh-session
> cat foo/main.ts
import { foo } from "./foo.js";

foo();

> cat foo/foo.js
export function foo() {
  throw new Error();
}

> phylum extension run -y ./foo
Extension error: Uncaught Error
  throw new Error();
        ^
    at foo (file:///Users/kyle/code/cli/foo/foo.js:2:9)
    at file:///Users/kyle/code/cli/foo/main.ts:2:1

> ./target/debug/phylum extension run -y ./foo
Extension error: Uncaught Error
    at foo (file:///Users/kyle/code/cli/foo/foo.js:2:9)
    at file:///Users/kyle/code/cli/foo/main.ts:3:1
```